awx/main/tasks/host_indirect.py

```diff
@@ -45,22 +45,30 @@ def build_indirect_host_data(job: Job, job_event_queries: dict[str, dict[str, st
     facts_missing_logged = False
     unhashable_facts_logged = False
 
+    job_event_queries_no_module = {}
+    for query_k, query_v in job_event_queries.items():
+        if len(parts := query_k.split('.')) != 3:
+            logger.info(f"Skiping malformed query '{query_k}'. Expected to be of the form 'a.b.c'")
```

Comment:
I _am_ concerned that someone runs a job that runs a net of 1 million host-tasks, which all hit this.

My issue is not that the log is bad, it's that the log would log 1 million times. I'd like to add some protection for this. I did this for some other logs.

Reply:
We don't loop over the job events until `for event in job.job_events.filter(event_data__isnull=False).iterator():`

This part is looping over the event queries, which should be relatively small.

Reply:
Good point here.

But you still have a log

```
logger.info(f"Malformed invocation module name '{resolved_action}'. Expected to be of the form 'a.b.c'")
```

seems that could still cause it to freak out.

Reply:
These event queries should be well formed and we want to know when they are not. I can’t think of an alternative to logging every case.

Reply:
> I can’t think of an alternative to logging every case.

Log the first case?

This runs once per job. When another job runs, you'll get another log.

Reply:
I kept _this_ log message. The number of times it will trigger is relative to the number of malformed queries, this should be small.

I changed the per-event log message to debug. Printing it only once is misleading as many events could be malformed.